table/block_based_table_reader.cc

```diff
@@ -153,8 +153,9 @@ class BlockBasedTable::IndexReader {
   virtual ~IndexReader() {}
 
   // Create an iterator for index access.
-  // An iter is passed in, if it is not null, update this one and return it
-  // If it is null, create a new Iterator
+  // If a non-null iter is passed in it MIGHT be used instead of creating a new
```

Comment:
Previously it would always make use of the passed BlockIter object. Currently the existing classes that inherit from index reader still make use of it. However the new class that is added with this patch, index reader for partitioned indexes, does not. So we needed to update the contract and make the users to check the return value to verify whether the input object was used by NewIterator or it was ignored.